### PR TITLE
(Fix) Visibility of private task assign to me

### PR DESCRIPTION
**Pull request by Lainow**
## Checklist before requesting a review

*Please delete options that are not relevant.*

- [x] I have read the CONTRIBUTING document.
- [x] I have performed a self-review of my code.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] This change requires a documentation update.

## Description

- It fixes !34489

I can see a private spot that has been assigned to me even though I don't have the right to see private spots.
Added the right to view my group's tasks.

### CODE REVIEW

File: src/CommonITILObject.php [hunk 1]
```diff
@@ -7053,6 +7053,7 @@ public function getTimelineItems(array $options = [])
                     'OR' => [
                         'is_private' => 0,
                         'users_id'   => Session::getCurrentInterface() === "central" ? (int)Session::getLoginUserID() : 0,
+                        'users_id_tech' => Session::getCurrentInterface() === "central" ? (int)Session::getLoginUserID() : 0,
```
Comment (Rom1-B) [inline src/CommonITILObject.php:7056]: Can you please add some tests ?

**Review by Rom1-B — CHANGES_REQUESTED** [review]

File: phpunit/functional/TicketTest.php [hunk 2]
```diff
@@ -6416,7 +6416,27 @@ protected function timelineItemsProvider(): iterable
...
+        ]);
+
+        $new_user_id = $tech_user->clone();
+        $user = new \User();
+        $user->update([
+            'id' => $new_user_id,
+            'login' => 'new_user',
+            'password' => 'new_user',
+            'profiles_id' => $new_tech
+        ]);
```
Comment (Rom1-B) [inline phpunit/functional/TicketTest.php:6439]: I'm not sure the clone is essential?
```suggestion
        $this->createItem('User', [
            'name' => 'new_user',
            'login' => 'new_user',
            'password' => 'new_user',
            'password2' => 'new_user',
            '_profiles_id' => $new_tech,
            '_entities_id' => getItemByTypeName('Entity', '_test_root_entity', true),
        ]);
```

File: phpunit/functional/TicketTest.php [hunk 3]
```diff
@@ -6416,7 +6416,27 @@ protected function timelineItemsProvider(): iterable
...
+
+        $tech_p = getItemByTypeName(\Profile::class, 'Technician');
+        $new_tech = $tech_p->clone();
+
+        $profileright = new \ProfileRight();
+        $profileright->update([
+            'profiles_id' => $new_tech,
+            'name'        => 'task',
+            'rights'      => 5137
+        ]);
```
Comment (Rom1-B) [inline phpunit/functional/TicketTest.php:6430]: You should create the profile directly without cloning, with only the rights you need.

**Review by Rom1-B — CHANGES_REQUESTED** [review]

File: phpunit/functional/TicketTest.php [hunk 4]
```diff
@@ -6418,6 +6418,19 @@ protected function timelineItemsProvider(): iterable
...
+                'name' => 'No Private Task',
+                'interface' => 'central',
+            ]
+        );
+        $this->createItem(\Profile::class, [
+            'name' => 'No Private Task',
+            'interface' => 'central',
+            '_entities_id' => getItemByTypeName('Entity', '_test_root_entity', true),
+        ]);
+
```
Comment (Rom1-B) [inline phpunit/functional/TicketTest.php:6433]: It creates the profile twice and none of them seem to be used
```suggestion
```

**Review by Rom1-B — CHANGES_REQUESTED** [review]

File: phpunit/functional/TicketTest.php [hunk 5]
```diff
@@ -6496,6 +6496,17 @@ protected function timelineItemsProvider(): iterable
             ]
         );
 
+        $this->createItem(
+            \TicketTask::class,
+            [
+                'tickets_id'    => $ticket->getID(),
+                'content'       => 'private task of only see his task user',
```
Comment (Rom1-B) [inline phpunit/functional/TicketTest.php:6503]:
```suggestion
                'content'       => 'private task assigned to normal user',
```

File: phpunit/functional/TicketTest.php [hunk 6]
```diff
@@ -6509,6 +6520,7 @@ protected function timelineItemsProvider(): iterable
             ],
             'expected_tasks'     => [
                 'private task of normal user',
+                'private task of only see his task user',
```
Comment (Rom1-B) [inline phpunit/functional/TicketTest.php:6523]:
```suggestion
                'private task assigned to normal user',
```

File: phpunit/functional/TicketTest.php [hunk 7]
```diff
@@ -6526,6 +6538,7 @@ protected function timelineItemsProvider(): iterable
...
```
Comment (Rom1-B) [inline phpunit/functional/TicketTest.php:6541]:
```suggestion
                'private task assigned to normal user',
```

File: phpunit/functional/TicketTest.php [hunk 8]
```diff
@@ -6561,6 +6574,7 @@ protected function timelineItemsProvider(): iterable
                 ],
                 'expected_tasks'     => [
                     'private task of normal user',
+                    'private task of only see his task user',
```
Comment (Rom1-B) [inline phpunit/functional/TicketTest.php:6577]:
```suggestion
                    'private task assigned to normal user',
```

**Review by Rom1-B — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Review by AdrienClairembault — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Comment by alisher372** [comment]
whats about assigned group to the private task

**Pull request merged**